### PR TITLE
feat: Minimal Hugging Face Image Classification Pipeline

### PR DESCRIPTION
**Pull request by jjerphan**
### Context

Prototype for https://github.com/flojoy-ai/studio/issues/839.

### Description

- [x] I've included a concise description of what each node does

### Styleguide

- [x] My node adheres to the [styleguide](https://github.com/flojoy-io/nodes/blob/main/node_styleguide.md) for Flojoy nodes

### Docs

- [ ] I've submitted a PR for a documentation page for the new node(s) that contains usage examples (see docs.flojoy.io)

### Testing

- [x] This PR includes a unit test ([example here](https://github.com/flojoy-io/nodes/blob/c28808f2d0aec8f116cdb3fccb46c4f5256574e9/TRANSFORMERS/ARITHMETIC/ADD/ADD_test_.py) and/or ideally a screenshot of the node's output on an example app.

### CODE REVIEW

**Comment by jjerphan** [comment]
The `create-docs-pr / send-pull-requests` CI check does not pass because, [as stated in the documentation](https://docs.github.com/en/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow):

> With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository.

I will then create PR on the repository directly.

File: AI_ML/IMAGE_CLASSIFICATION/HUGGING_FACE_PIPELINE/HUGGING_FACE_PIPELINE.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    import os
+    import numpy as np
+    import pandas as pd
+    import PIL.Image as PILImage
+    from flojoy.utils import FLOJOY_CACHE_DIR
+    from typing import List, Dict
+
+    # Setting transformers cache directory to flojoy cache directory before importing transformers
+    # not to pollute the user's cache directory.
+    os.environ["TRANSFORMERS_CACHE"] = os.path.join(FLOJOY_CACHE_DIR, "transformers")
```
Comment (jjerphan) [inline AI_ML/IMAGE_CLASSIFICATION/HUGGING_FACE_PIPELINE/HUGGING_FACE_PIPELINE.py:51]: We probably might want to make this systematic.

Comment (Roulbac) [reply AI_ML/IMAGE_CLASSIFICATION/HUGGING_FACE_PIPELINE/HUGGING_FACE_PIPELINE.py:51]: +1

We previously wrapped the `huggingface-hub` library within `flojoy` though it's simpler to just ensure `os.environ["HF_HOME"]` is set whenever flojoy is imported

**Review by Roulbac — COMMENTED** [review]
LGTM, can you however please share to what's the issue with this node on Windows?

File: AI_ML/IMAGE_CLASSIFICATION/HUGGING_FACE_PIPELINE/HUGGING_FACE_PIPELINE_test_.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    assert isinstance(df_classification_confidence_scores, DataFrame)
+
+    first_class_confidence_scores = df_classification_confidence_scores.m.iloc[0, :]
+
+    assert first_class_confidence_scores["label"] == "overskirt"
+    assert first_class_confidence_scores["score"] > 0.725
+
+
+# Skip this test on Windows
+@pytest.mark.skipif(sys.platform == "win32", reason="does not run on windows")
```
[the remark this answers is not in the document]
Comment (jjerphan) [reply AI_ML/IMAGE_CLASSIFICATION/HUGGING_FACE_PIPELINE/HUGGING_FACE_PIPELINE_test_.py:55]: This test is slow, and also crashes on Windows and I need to understand why.

**Comment by jjerphan** [comment]
The test failures on `main` are being addressed by https://github.com/flojoy-ai/nodes/pull/268.

File: AI_ML/IMAGE_CLASSIFICATION/HUGGING_FACE_PIPELINE/HUGGING_FACE_PIPELINE.py (hunk 1 above)
Comment (jjerphan) [reply AI_ML/IMAGE_CLASSIFICATION/HUGGING_FACE_PIPELINE/HUGGING_FACE_PIPELINE.py:51]: Can we add it to Flojoy's processes' environment?

Comment (Roulbac) [reply AI_ML/IMAGE_CLASSIFICATION/HUGGING_FACE_PIPELINE/HUGGING_FACE_PIPELINE.py:51]: https://github.com/flojoy-ai/python/pull/97 Incoming 👍

File: AI_ML/IMAGE_CLASSIFICATION/HUGGING_FACE_PIPELINE/HUGGING_FACE_PIPELINE_test_.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Roulbac) [reply AI_ML/IMAGE_CLASSIFICATION/HUGGING_FACE_PIPELINE/HUGGING_FACE_PIPELINE_test_.py]: In `nodes` there is a set of slow tests we run on a nightly basis to keep the continuous CI which runs on every commit flowing smoothly

**Review by Roulbac — APPROVED** [review]

**Pull request merged**